### ISSUE TITLE
All matching messages are treated as warnings even if they're logged at a lower level

### ISSUE THREAD

**Issue by wilkinsona** [asciidoctor/asciidoctor-gradle-plugin#529]
I'd like to be able to configure an Asciidoctor task to fail when any warning (or higher) log records are handled. I do not want log records at info level or lower to be considered. My expectation was that `fatalWarnings(".*")` would achieve this goal as it would consider any warnings that match `.*` to be fatal. The behaviour that I am seeing is that any logging, irrespective of its level, is considered to be fatal.

Looking at the code in `ExecutorBase`, the `LogHandler` doesn't consider the level of the `LogRecord` when passing its message to `addMatchingMessage`:

https://github.com/asciidoctor/asciidoctor-gradle-plugin/blob/8d324c24c37e6c649bde038257a2a456e008368f/jvm/src/main/groovy/org/asciidoctor/gradle/remote/ExecutorBase.groovy#L152-L173

This feels like a bug to me as the API suggests that it is warning messages that are being made fatal, not all messages.

FWIW, the Asciidoctor Maven Plugin allows level-based failures to be configured. For example, the following configuration will cause the execution of the goal to fail if any INFO level of higher messages are logged.:

```xml
<configuration>
	<logHandler>
		<outputToConsole>false</outputToConsole>
		<failIf>
			<severity>INFO</severity>
		</failIf>
	</logHandler>
</configuration>
```

**Comment by ysb33r** [comment]
That should be an easy enhancement. Probably we can just do 
```
fatalWarnings ~/foo/, 'WARN'
```
If not specified then it considers all warnings.

**Comment by ysb33r** [comment]
Should be fixed by #536 and released as part of 3.2.0.
